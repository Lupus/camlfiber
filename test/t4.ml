[@@@warning "-52"]

let print_exc e = print_string (Printexc.to_string e);
                  print_newline ()

let _ =
  let r = ref (Fiber.create (fun () -> ()) ()) (* dummy placeholder *) in
  let f = Fiber.create (fun () ->
              try Fiber.resume !r (* fiber recursion is not allowed *)
              with Invalid_argument("Fiber.resume") as e -> print_exc e) () in
  r := f;
  Fiber.resume f
