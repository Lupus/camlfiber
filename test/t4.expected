Invalid_argument("Fiber.resume")
